searchlib/src/vespa/searchlib/tensor/angular_distance.cpp

```diff
@@ -49,4 +50,74 @@ AngularDistance::calc(const vespalib::eval::TypedCells& lhs,
 template class AngularDistanceHW<float>;
 template class AngularDistanceHW<double>;
 
+
+template<typename FloatType>
+class BoundAngularDistance : public BoundDistanceFunction {
+private:
+    const vespalib::hwaccelrated::IAccelrated & _computer;
+    mutable TemporaryVectorStore<FloatType> _tmpSpace;
+    const vespalib::ConstArrayRef<FloatType> _lhs;
+    double _lhs_norm_sq;
+public:
+    BoundAngularDistance(const vespalib::eval::TypedCells& lhs)
+        : BoundDistanceFunction(vespalib::eval::get_cell_type<FloatType>()),
+          _computer(vespalib::hwaccelrated::IAccelrated::getAccelerator()),
+          _tmpSpace(lhs.size),
+          _lhs(_tmpSpace.storeLhs(lhs))
+    {
+        auto a = &_lhs[0];
+        _lhs_norm_sq = _computer.dotProduct(a, a, lhs.size);
+    }
+    double calc(const vespalib::eval::TypedCells& rhs) const override {
+        size_t sz = _lhs.size();
+        vespalib::ConstArrayRef<FloatType> rhs_vector = _tmpSpace.convertRhs(rhs);
```

Comment:
This assumes that the BoundDistanceFunction is only used by a single thread. That requirement should probably be mentioned in the class description of BoundDistanceFunction.